cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/BasicPluginInfo/index.tsx

```diff
@@ -55,13 +55,31 @@ const BasicPluginInfoView: React.FC<ICreateContext & WithStyles<typeof styles>>
   jsonView,
   setJsonView,
   outputName,
+  setPluginState,
 }) => {
   const [localPluginName, setLocalPluginName] = React.useState(pluginName);
   const [localPluginType, setLocalPluginType] = React.useState(pluginType);
   const [localDisplayName, setLocalDisplayName] = React.useState(displayName);
   const [localEmitAlerts, setLocalEmitAlerts] = React.useState(emitAlerts);
   const [localEmitErrors, setLocalEmitErrors] = React.useState(emitErrors);
 
+  // In case user uploads new file
+  React.useEffect(() => {
+    setLocalPluginName(pluginName);
```

Comment:
This won't be scalable. If the user uploads a new file there are two things we can do,

1. Validate the JSON and if it has all the right widget and attribute references
2. Set the output of step #1 to the state and allow the UI now re-render based on the state change.

Reply:
Resolved.

Whenever the `JSONStatus` changes to success state, the UI will unmount and remount. This has been enabled by adding a loading state in https://github.com/cdapio/cdap/blob/86724e9a3695e559255c8c8ec7a1b97288dff6a0/cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/index.tsx